### PR TITLE
CBG-1532: Unregister prometheus stats on db close and disable for tests

### PR DESCRIPTION
**Pull request by JRascagneres**
All the work done here is around properly tearing down prometheus stats on a database close. Previously we were only removing our internal reference to them but they were still initialized in Prometheus.

To properly remove stats we need to 'Unregister' them. 

In order to properly 'Unregister' them they require a 'Desc' which acts as a unique identifier for each stat. Therefore inside of each of the previously unused Describe methods we now send the stats description over the channel.

Our current descriptions were built slightly incorrectly as our labels were being provided through the 'variableLabels' parameter rather than the 'constLabels'. Due to that fact that a metrics labels will not change during its lifetime they should be provided inside of 'constLabels'.

Additionally to avoid test issues an additional variable `SkipPrometheusStatsRegistration` has been implemented. When this is set to true (as is the default case for tests via TestMain) the stat initializations will not register with Prometheus. As `NewSyncGatewayStats()` is called from `init()` this is only applicable to non-global stats but this is fine as if any tests re-initialize those stats those further initializations will be skipped.

To validate this a test has been added which tests:
- Creation of db stats
- Removal of db stats
- Re-Creation of db stats under same name
- Adding a second db

Therefore in summary changes are:
- Modify stat descriptions
- Return descriptions in Describe()
- Unregister stats on ClearDBStats()
- Unit test ability to skip prometheus registration of stats
- Unit test to test functionality

### CODE REVIEW

**Review by bbrks — CHANGES_REQUESTED** [review]
Approach seems reasonable to me, but have a question about the changes in `TestUnsupportedConfig`.

File: rest/api_test.go [hunk 1]
```diff
@@ -3994,6 +3994,7 @@ func TestUnsupportedConfig(t *testing.T) {
 
 	_, err = sc.AddDatabaseFromConfig(&testProviderOnlyConfig)
 	assert.NoError(t, err, "Error adding testProviderOnly database.")
+	sc._removeDatabase("test_provider_only")
```
[the remark this answers is not in the document]
Comment (JRascagneres) [reply rest/api_test.go]: Yeah those aren't needed, my bad, was left over from a previous attempt at fixing the issue.

**Review by bbrks — COMMENTED** [review]
I don't think we have automated testing for Prometheus - have you given the moved ReplicatorStats a quick sanity check?

**Comment by JRascagneres** [comment]
> I don't think we have automated testing for Prometheus - have you given the moved ReplicatorStats a quick sanity check?

Validated that adding replications has them successfully collected

**Review by bbrks — APPROVED** [review]

**Pull request merged**